### ISSUE TITLE
Product is not published after being published

### ISSUE THREAD

**Issue by AlmogCohen** [mirumee/saleor-dashboard#388]
### What I'm trying to achieve
Publish a digital product

### Steps to reproduce the problem
1. Go to the products page (http://localhost:9000/products/?asc=true&sort=name)
2. Choose two products that at least one of them is not published (we select two products due to the bug reported in #387)
3. Click "publish"
4. Refresh the page and see that the product is not published
5. I've got a feeling like this might be not a bug in the dashboard...as all requests seem fine

### What I expected to happen
The published product to be published

### Screenshots
Recording can be found here: https://share.getcloudapp.com/yAuLqzv4
In the recording you can see:
1. The product being published and remain unpublished
2. The product publish GRAPHQL request
3. The product as it is returned from the GraphQL product list

**System information**
Operating system: Docker over OSX
Browser: Latest Chrome

**Comment by AlmogCohen** [comment]
This might be a backend bug

**Comment by karolkielecki** [comment]
Hey @AlmogCohen. Thanks for reporting this issue! It seems this bug is related to the master branch. We're working to find a solution, but I'm not sure when it will be fixed. I will keep you updated.

We recommend you to use Dashboard 2.0 and Saleor 2.9 (the latest stable version). If you need any help, please let me know.

**Comment by AlmogCohen** [comment]
Thanks @karolkielecki , I'll give it now a try with 2.0 and 2.9 and hopefully things will work better.

I'd like just to note that the official installation guide for all operating systems does *NOT* mention how to install the specific versions using the release tag ([for example](https://saleor.readthedocs.io/en/stable/gettingstarted/installation-linux.html#installation)). I assume it can mislead many newcomers to the project.

I know how to do it now that you mentioned that :)

**Comment by AlmogCohen** [comment]
@karolkielecki I've just confirmed this bug is also present on 2.0 + 2.9

I've drilled down into this issue, these are the results:
*TL;DR:* More of a FE UX flaw than an actual backend bug.

1. The client successfully ask for bulk publish of products
2. The Saleor backend does make the product `is_available=True` successfully.
3. Then the client asks for the product list.
4. The product I published is still marked as "not published". When I look I understand that [this line](https://github.com/mirumee/saleor/blob/952d91348583902bd030070e33d6f89464c00482/saleor/product/models.py#L328) returns `False` as the product is not in stock, although it is available.
5. The frontend shows "Not Published" https://share.getcloudapp.com/bLuGng5Y

Conclusion: This is confusing! The product IS published, but is marked as "Not Published" because it is out of stock....how could I possibly guess that using the FE? Since I just publish the product, it is frustrating to see it remain unpublished.

Maybe the bug now can be renamed to "Product marked as not published when out of stock". Of course it is not published, but the user shouldn't be guessing what is the reason for that. On top of that, "out of stock" is the default value for a new product, so this bug is even more likely  to occur. Is it a UX bug? a real bug? 

I hope you do understand that I have no intensions to criticize but to encourage and support making this product even better! I think that a better behavior here would be to show "Out of Stock" rather than "Not Published".

I'd love to hear your thoughts!

**Comment by AlmogCohen** [comment]
Indeed, after changing the product stock to 1 and refreshing the page, the product is not "published"

**Comment by maarcingebala** [comment]
I think this might be an issue in Saleor 2.9 / Dashboard 2.0. On `master` of both projects, we have a yet different situation because we recently added warehouses backend (https://github.com/mirumee/saleor/pull/4986), while the UI doesn't support it yet. E.g. for products in `populatedb` we create multiple warehouses, but the UI only allows us to change quantity in one of them, so it's really hard to test now. I would wait with further debugging until we add UI (#390).
